controller/eks-cluster-config-handler.go

```diff
@@ -444,114 +445,114 @@ func (h *Handler) validateCreate(ctx context.Context, config *eksv1.EKSClusterCo
 		}
 		if config.Spec.PublicAccessSources == nil {
-			return fmt.Errorf(cannotBeNilError, "publicAccessSources", config.Name)
+			return fmt.Errorf(cannotBeNilError, "publicAccessSources", config.Spec.DisplayName, config.Name)
 		}
 	}
 	for _, ng := range config.Spec.NodeGroups {
-		cannotBeNilError := "field [%s] cannot be nil for nodegroup [%s] in non-nil cluster [%s]"
+		cannotBeNilError := "field [%s] cannot be nil for nodegroup [%s] in non-nil cluster [%s (id: %s)]"
 		if !config.Spec.Imported {
 			if ng.LaunchTemplate != nil {
 				if ng.LaunchTemplate.ID == nil {
-					return fmt.Errorf(cannotBeNilError, "launchTemplate.ID", *ng.NodegroupName, config.Name)
+					return fmt.Errorf(cannotBeNilError, "launchTemplate.ID", *ng.NodegroupName, config.Spec.DisplayName, config.Name)
 				}
 				if ng.LaunchTemplate.Version == nil {
-					return fmt.Errorf(cannotBeNilError, "launchTemplate.Version", *ng.NodegroupName, config.Name)
+					return fmt.Errorf(cannotBeNilError, "launchTemplate.Version", *ng.NodegroupName, config.Spec.DisplayName, config.Name)
 				}
 			} else {
 				if ng.Ec2SshKey == nil {
-					return fmt.Errorf(cannotBeNilError, "ec2SshKey", *ng.NodegroupName, config.Name)
+					return fmt.Errorf(cannotBeNilError, "ec2SshKey", *ng.NodegroupName, config.Spec.DisplayName, config.Name)
 				}
 				if ng.ResourceTags == nil {
-					return fmt.Errorf(cannotBeNilError, "resourceTags", *ng.NodegroupName, config.Name)
+					return fmt.Errorf(cannotBeNilError, "resourceTags", *ng.NodegroupName, config.Spec.DisplayName, config.Name)
 				}
 				if ng.DiskSize == nil {
-					return fmt.Errorf(cannotBeNilError, "diskSize", *ng.NodegroupName, config.Name)
+					return fmt.Errorf(cannotBeNilError, "diskSize", *ng.NodegroupName, config.Spec.DisplayName, config.Name)
 				}
 				if !aws.ToBool(ng.RequestSpotInstances) && ng.InstanceType == "" {
-					return fmt.Errorf(cannotBeNilError, "instanceType", *ng.NodegroupName, config.Name)
+					return fmt.Errorf(cannotBeNilError, "instanceType", *ng.NodegroupName, config.Spec.DisplayName, config.Name)
 				}
 				if aws.ToBool(ng.Arm) && ng.InstanceType == "" {
-					return fmt.Errorf(cannotBeNilError, "instanceType", *ng.NodegroupName, config.Name)
+					return fmt.Errorf(cannotBeNilError, "instanceType", *ng.NodegroupName, config.Spec.DisplayName, config.Name)
 				}
 			}
 			if ng.NodegroupName == nil {
-				return fmt.Errorf(cannotBeNilError, "name", *ng.NodegroupName, config.Name)
+				return fmt.Errorf(cannotBeNilError, "name", *ng.NodegroupName, config.Spec.DisplayName, config.Name)
 			}
 			if nodeP[*ng.NodegroupName] {
-				return fmt.Errorf("node group names must be unique within the [%s] cluster to avoid duplication", config.Name)
+				return fmt.Errorf("node group name [%s] must be unique within the [%s (id: %s)] cluster to avoid duplication", *ng.NodegroupName, config.Spec.DisplayName, config.Name)
```

Comment:
```suggestion
				return fmt.Errorf("node group names must be unique within the [%s (id: %s)] cluster to avoid duplication", config.Spec.DisplayName, config.Name)
```

Reply:
To be consistent with gke-operator, and to have a more clear error, I have modified the error message to be `node group name [%s] is not unique within the ....`.